src/openforms/js/components/admin/form_design/utils.js

```diff
@@ -65,9 +65,9 @@ const updateKeyReferencesInLogic = (existingLogicRules, originalKey, newKey) =>
     let newRule = {...rule};
     // Replace the key in the JSON trigger
     const stringJsonTrigger = JSON.stringify(rule.jsonLogicTrigger);
-    const compToReplace = JSON.stringify({var: originalKey});
+    const patternToReplace = new RegExp(`\{"var":"${originalKey}(\\.)?(.+?)?"\}`, 'g');
     newRule.jsonLogicTrigger = JSON.parse(
-      stringJsonTrigger.replace(compToReplace, JSON.stringify({var: newKey}))
+      stringJsonTrigger.replaceAll(patternToReplace, `\{"var":"${newKey}$1$2"\}`)
```

Comment:
how does this behave with multiple vars (since regex is usually greedy)?

if you have a rule with `[{"var": "foo.bar"}, {"var": "foo.baz"}]`, is this going to replace every var individually, or will it consider the string `{"var": "foo.bar"}, {"var": "foo.baz"}`  to be one occurrence?

Perhaps the `$2` group should be limited to `[0-9a-zA-Z_\-]`?

Reply:
It would give `[{"var":"fooNew.bar"}, {"var":"fooNew.baz"}]` (the `+?` in the 2nd group makes it match as few times as possible). But I think it's a good idea to limit it to the chars you mention.